src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if ((target instanceof Herbivore) && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+      }
+    }
+  }
```

Comment:
The `bite` method currently filters the `Animal.alive` array every time a Carnivore bites a Herbivore and its health drops below 0. This is inefficient because it will iterate over the entire array even if no animal has died. Consider moving the filtering logic to a separate method and call it only when an animal's health actually reaches 0 to improve performance.